### PR TITLE
Fix namespace parsing regex to allow both allowed forms

### PR DESCRIPTION
**Pull request by akcoder**
Allow namespace form of "namespace My\Name\Space {" in addition to "namespace My\Name\Space;"

### CODE REVIEW

**Comment by akcoder** [comment]
Is there any chance of getting this merged?

File: DependencyInjection/Compiler/AnnotationConfigurationPass.php [hunk 1]
```diff
@@ -109,7 +109,7 @@ private function getClassName($filename)
     {
         $src = file_get_contents($filename);
 
-        if (!preg_match('/\bnamespace\s+([^;]+);/s', $src, $match)) {
+        if (!preg_match('/\bnamespace\s+([^;\{\s]+)\s*?[;\{]/s', $src, $match)) {
```
Comment (Ener-Getick) [inline DependencyInjection/Compiler/AnnotationConfigurationPass.php]: the ``?`` after ``\s`` doesn't seem to be needed.

**Comment by Ener-Getick** [comment]
Otherwise fine for me :+1:

File: DependencyInjection/Compiler/AnnotationConfigurationPass.php (hunk 1 above)
Comment (stof) [inline DependencyInjection/Compiler/AnnotationConfigurationPass.php]: the ``?`` here is about making the ``*`` quantifier non-greedy. It is *not* the same ``?`` than the one making a character optional: http://php.net/manual/fr/regexp.reference.repetition.php

Comment (Ener-Getick) [inline DependencyInjection/Compiler/AnnotationConfigurationPass.php]: Ah right sorry I didn't know that.

**Comment by Ener-Getick** [comment]
@stof do you see anything else to chage?

**Comment by Ener-Getick** [comment]
Thank you @akcoder

**Comment by akcoder** [comment]
Thank you for merging this! Now I don't have to apply my patch every time I update :)

**Pull request merged**